src/sentry/tasks/relay.py

```diff
@@ -114,15 +115,26 @@ def should_update_cache(organization_id=None, project_id=None, public_key=None)
     )
 
 
-def schedule_update_config_cache(
-    generate, project_id=None, organization_id=None, public_key=None, update_reason=None
-):
+def schedule_update_config_cache(*args, **kwargs):
     """
     Schedule the `update_config_cache` with debouncing applied.
 
     See documentation of `update_config_cache` for documentation of parameters.
     """
 
+    # schedule_update_config_cache may be called from model hooks during an
+    # open transaction. In that case, wait until the current transaction has
+    # been committed or rolled back to ensure we don't read stale data in the
+    # task.
+    #
+    # If there is no transaction open, on_commit should run immediately.
+    transaction.on_commit(lambda: _schedule_update_config_cache_impl(*args, **kwargs))
```

Comment:
These hooks will be running inside the transaction, so would have the same issue

Reply:
@wedamija floris was just suggesting to implement the fix in `post_save`

@flub my first solution was exactly that, but then i thought it would be better if the user of this function didn't have to think about transactions at all, i.e. we handle this `1` times instead of `n`.

i think changing the behavior of all post_save hooks to run after the transaction has completed would solve this as well, for all signal handlers ever

Reply:
I don't think changing the behaviour of all `post_save` hooks to run outside of a transaction is the solution here. We likely rely on this for transactional integrity in some places. I think that writing something that allows us to detect tasks fired within a transaction would probably be enough to catch issues like this.